trie-db/src/node.rs

```diff
@@ -577,29 +638,67 @@ impl NodePlan {
 				value.as_mut(),
 		}
 	}
+
+	/// Check if the node has a location for value.
+	pub fn has_location_value(&self) -> bool {
+		self.value_plan().map(|v| !v.is_inline()).unwrap_or(false)
+	}
+
+	/// Check how many children location value node has.
+	pub fn nb_location_children(&self) -> usize {
+		match self {
+			NodePlan::Extension { child: NodeHandlePlan::Hash(_), .. } => 1,
+			NodePlan::Branch { children, .. } | NodePlan::NibbledBranch { children, .. } => {
+				let mut count = 0;
+				for child in children {
+					if let Some(NodeHandlePlan::Hash(_)) = child {
+						count += 1;
+					}
+				}
+				count
+			},
+			_ => 0,
+		}
+	}
+
+	pub fn attached_change_set_location<L: Copy + Default>(&self, locations: &[L]) -> Option<L> {
```

Comment:
I did extract this to be able to still work on nodeplan up to a certain extent (changes from arkpar/new-trie in iterator.rs when decoding nodes), but I think location should just be included into nodeplan